ipn/ipnlocal/extension_host.go

```diff
@@ -466,37 +435,20 @@ func (h *ExtensionHost) DetermineBackgroundProfile(profiles ipnext.ProfileStore)
 	return ipn.LoginProfileView{}
 }
 
-// RegisterControlClientCallback implements [ipnext.Host].
-func (h *ExtensionHost) RegisterControlClientCallback(cb ipnext.NewControlClientCallback) {
-	if h != nil {
-		addFuncHook(h, &h.newControlClientCbs, cb)
-	}
-}
-
 // NotifyNewControlClient invokes all registered control client callbacks.
 // It returns callbacks to be executed when the control client shuts down.
 func (h *ExtensionHost) NotifyNewControlClient(cc controlclient.Client, profile ipn.LoginProfileView) (ccShutdownCbs []func()) {
 	if !h.active() {
 		return nil
 	}
-	if len(h.newControlClientCbs) > 0 {
-		ccShutdownCbs = make([]func(), 0, len(h.newControlClientCbs))
```

Comment:
I removed this one because control client changes are so rare & so costly already, that this is a drop in the bucket